packages/rum/src/rum.ts

```diff
@@ -310,7 +346,7 @@ function makeRumEventHandler(
       if (session.isTracked() && view && view.sessionId) {
         const action = parentContexts.findAction(startTime)
         const rumEvent = assemble(event, { action, view, rum: rumContextProvider() })
-        const message = combine(globalContextProvider(), customerContext, withSnakeCaseKeys(rumEvent))
+        const message = combine(customerContext || globalContextProvider(), withSnakeCaseKeys(rumEvent))
```

Comment:
What is the reasoning for that? reverting this change does not fail any unit test

Reply:
I added a unit test for it. The goal is to make sure the current global context is not used when formating a user action that was sent a while ago (same as logs). But this could be challenged... WDYT?

Reply:
ok I missed the [early combine](https://github.com/DataDog/browser-sdk/pull/551/files#diff-fcb436c4fbf0511b343fcf0e467d005bR102) of action and global context.

FMU, it is not the same strategy as logs since in the logs case we are saving both [LogsMessage and CurrentContext](https://github.com/DataDog/browser-sdk/blob/master/packages/logs/src/logs.entry.ts#L115) to decide how to combine them later.

By doing the same for RUM, we could pass the global context in the handler and keep all the combine logic at a single place:

```ts
const message = combine(savedGlobalContext || globalContextProvider(), customerContext, withSnakeCaseKeys(rumEvent))
```

which IMO would be easier to follow/understand.

Reply:
Okay! Sure, it's not the same strategy, what I meant was the end result is the same (using the old global context instead of the current one).

In logs, we have to keep the message and current context separated because we have to set the current RUM context in between. This is not needed in RUM, but if you think it's easier to understand, I'll store the current context and customer context separated